corehq/apps/reports/views.py

```diff
@@ -1937,20 +1937,26 @@ def archive_form(request, domain, instance_id):
 
 
 def _get_cases_with_forms_message(domain, cases_with_other_forms, case_id_from_request):
-    def _get_case_link(case_id, name):
-        if case_id == case_id_from_request:
-            return _("%(case_name)s (this case)") % {'case_name': name}
-        else:
-            return '<a href="{}#!history">{}</a>'.format(reverse('case_data', args=[domain, case_id]), name)
+    def _get_all_case_links():
+        all_case_links = []
+        for case_id, case_name in cases_with_other_forms.items():
+            if case_id == case_id_from_request:
+                all_case_links.append([(
+                    format_html(_("{}s (this case)"), case_name)
+                )])
+            else:
+                all_case_links.append([format_html(_(
+                    '<a href="{}#!history">{}</a>'),
+                    reverse("case_data", args=[domain, case_id]),
+                    case_name
+                )])
```

Comment:
If we leave in the brackets for `format_html`, then we'd need matching brackets for the `if` clause above. I think it's more intuitive to leave the brackets out and have `all_case_links` just be a list of links/names. We can create the necessary tuples with an in-line list-comprehension in `format_html_join`